lesson25/main.js

```diff
@@ -19,14 +32,84 @@ submitBtn.addEventListener(
   false
 );
 
+const checkInputVal = (targetForm) => {
+  const targetVal = targetForm.value;
+  let result;
+
+  switch (targetForm.id) {
+    case "userName":
+      const minLength = 1;
+      const maxLength = 15;
+      result = targetVal.length >= minLength && targetVal.length <= maxLength;
+      break;
+    case "email":
+      const emailValid = /^\w[\w_.-]*@.+\..+/;
+      result = emailValid.test(targetVal);
+      break;
+    case "pass":
+      const passValid = /(?=.*[a-z])(?=.*[A-Z])(?=.*\d)\w{8,}/;
+      result = passValid.test(targetVal);
+      break;
+    default:
+      console.error(`は見つかりませんでした`);
+  }
+
+  result ? (flags[targetForm.id] = true) : (flags[targetForm.id] = false);
+  return result;
+};
+
+const createErrorMessage = (targetFormId) => {
+  const errorTxt = createElementWithClassName("p", "error-txt");
+  errorTxt.id = "js-errorTxt";
+
+  switch (targetFormId) {
+    case "userName":
+      errorTxt.textContent = "ユーザー名は15文字以下にしてください。";
+      break;
+    case "email":
+      errorTxt.textContent = "メールアドレスの形式になっていません。";
+      break;
+    case "pass":
+      errorTxt.textContent =
+        "8文字以上の大小の英数字を交ぜたものにしてください。";
+      break;
+    default:
+      console.error(`は見つかりませんでした`);
+  }
+
+  return errorTxt;
+};
+
+const renderErrorMessage = (targetFormId) => {
+  const errorTxt = createErrorMessage(targetFormId);
+  const targetInput = document.getElementById(targetFormId);
+  targetInput.after(errorTxt);
+};
+
+//It's called in the HTML file
+const validateInputVal = (targetForm) => {
+  const check = checkInputVal(targetForm);
+  const errorTxtEle = targetForm.nextElementSibling;
+  !check && !errorTxtEle && renderErrorMessage(targetForm.id);
+  check && errorTxtEle && errorTxtEle.remove();
+  switchSbumitBtn();
+};
+
 const observer = new IntersectionObserver((targets) => {
   if (targets[0].intersectionRatio === 1) {
     checkBox.disabled = false;
     checkBox.checked = true;
+    flags.checkBox = true;
   }
+  switchSbumitBtn();
 });
```

Comment:
Even if the user unchecked the checkbox after it was checked, the submit button would remain active.
It will be much better if you add this code.

```js
checkBox.addEventListener(
  "click",
  (e) => {
    e.target.checked ? (flags.checkBox = true) : (flags.checkBox = false);
    switchSbumitBtn();
  },
  false
);
```

Reply:
Now check box works in the correct way. Thanks to you :)
[68febc0](https://github.com/yuka830/js-lessons/pull/30/commits/68febc0d15d995c632194995bc55eda7faf3d176)